spec/game_spec.rb

```diff
@@ -10,54 +20,58 @@ def initialize
   def output(message)
     @messages << message
   end
-
-  def gets_input
-    1
-  end
-end
-
-class TestDisplay
-  def present(board)
-    board.values
-  end
 end
 
-class TestPrompt
-  def call(_message, _error_message)
-    8
-  end
-end
-
-class TestBoard
-  attr_accessor :values
+class TestGameLooper
+  attr_reader :console
 
-  def initialize
-    @values = 'empty board'
+  def initialize(console)
+    @console = console
   end
 
-  def get_space(row, column); end
-
-  def mark_space(token, space)
-    @values = "board with an #{token} mark in space #{space}"
+  def loop
+    console.output('Looping')
   end
 end
 
 describe 'Game' do
   context '#run' do
     it 'plays the game' do
       console = TestConsole.new
-      welcome_message = 'Hello player!'
-      display = TestDisplay.new
-      prompt = TestPrompt.new
-      board = TestBoard.new
+      game_looper = TestGameLooper.new(console)
+      game = Game.new(console:, game_looper:)
 
-      game = Game.new(console, welcome_message, display, prompt, board)
+      game.run
+
+      expected_output = ['Welcome to Tic Tac Toe!', 'Looping', 'Thank you for playing. Goodbye!']
+      expect(console.messages).to eq(expected_output)
+    end
+  end
+
+  context 'when the board is full' do
+    it 'exits the game' do
```

Comment:
Very nice. Using all the real objects here, as opposed to fakes.

Reply:
Thanks for all the positive feedback, I appreciate it!